File: api-tester-java/src/main/java/com/revenuecat/apitesterjava/ProductTypeAPI.java
package com.revenuecat.apitesterjava;

import com.revenuecat.purchases.ProductType;

@SuppressWarnings({"unused"})
final class ProductTypeAPI {
    static void check(final ProductType type) {
        switch (type) {
            case SUBS:
            case INAPP:
            case UNKNOWN:
        }
    }
}

File: api-tester-kotlin/src/main/java/com/revenuecat/apitesterkotlin/ProductTypeAPI.kt
package com.revenuecat.apitesterkotlin

import com.revenuecat.purchases.ProductType

@Suppress("unused")
private class ProductTypeAPI {
    fun check(type: ProductType) {
        when (type) {
            ProductType.SUBS,
            ProductType.INAPP,
            ProductType.UNKNOWN
            -> {}
        }
    }
}
